src/shared/internal-calendar/week/week-day-cell.tsx

```diff
@@ -0,0 +1,136 @@
+import dayjs, { Dayjs } from "dayjs";
+import { CalendarHelper, DateHelper } from "../../../util";
+import { CellStyleProps, CellType, DayCell, DayCellProps } from "../day-cell";
+
+interface Props {
+    date: Dayjs;
+    calendarDate: Dayjs;
+    selectedDate: string;
+    hoverDate: string;
+    minDate?: string | undefined;
+    maxDate?: string | undefined;
+    disabledDates?: string[] | undefined;
+    allowDisabledSelection?: boolean | undefined;
+    onSelect: (value: Dayjs, disabled: boolean) => void;
+    onHover: (value: string, disabled: boolean) => void;
+}
+
+export const WeekDayCell = ({
+    date,
+    calendarDate,
+    selectedDate,
+    hoverDate,
+    minDate,
+    maxDate,
+    disabledDates,
+    allowDisabledSelection,
+    onSelect,
+    onHover,
+}: Props) => {
+    // =========================================================================
+    // CONSTS
+    // =========================================================================
+    const disabled = CalendarHelper.isDisabledDay(
+        date,
+        disabledDates,
+        minDate,
+        maxDate
+    );
+    const interactive = !disabled || allowDisabledSelection;
+    const { start: weekStart, end: weekEnd } = CalendarHelper.getWeekStartEnd(
+        DateHelper.toDayjs(selectedDate)
+    );
+    const { start: hoverStart, end: hoverEnd } = CalendarHelper.getWeekStartEnd(
+        DateHelper.toDayjs(hoverDate)
+    );
+
+    const isSelected =
+        selectedDate && date.isBetween(weekStart, weekEnd, "day", "[]");
+    const isHover =
+        hoverDate && date.isBetween(hoverStart, hoverEnd, "day", "[]");
+    const isStart =
+        (isSelected && date.isSame(weekStart)) ||
+        (isHover && date.isSame(hoverStart));
+    const isEnd =
+        (isSelected && date.isSame(weekEnd)) ||
+        (isHover && date.isSame(hoverEnd));
+
+    // =========================================================================
+    // EVENT HANDLERS
+    // =========================================================================
+    const handleSelect = () => {
+        onSelect(date, !interactive);
+    };
+
+    const handleHover = () => {
+        onHover(date.format("YYYY-MM-DD"), !interactive);
+    };
+
+    // =========================================================================
+    // HELPERS
+    // =========================================================================
+    const getRangeStyle = (): CellStyleProps => {
+        const props: CellStyleProps = {};
+
+        let type: CellType = undefined;
+        if (isSelected && isHover) {
+            type = "hover-current";
+            props.shadow = true;
+            props.circleShadow = isStart || isEnd;
+        } else if (isSelected) {
+            type = "selected-outline";
+        } else if (isHover) {
+            type = "hover-dash";
+        }
+
+        if (type) {
+            if (isStart) {
```

Comment:
could this be 2 `if-else` statements?

Reply:
do you mean this?

```
if (isStart)
else if (isEnd)

if (!isStart)
else if (!isEnd)
```

the second one won't work because`!isStart` and `!isEnd` are not mutually exclusive, so Mon-Fri need both bgLeft and bgRight

Reply:
no more like 
```
if (isStart) {
} else { }

if (isEnd) {
} else {}
```